src/paas_charm/app.py

```diff
@@ -17,6 +17,9 @@
 
 logger = logging.getLogger(__name__)
 
+if TYPE_CHECKING:
+    from paas_charm.redis import PaaSRedisRelationData
```

Comment:
I assume there is an issue if this is imported without `TYPE_CHECKING`?

Is there no other way to resolve this issue? For example, alway install the typing library for the redis module or something like that?

Reply:
Unfortunately, this code is imported and used as a lib and we don't quite have control over what libs are pre-installed by the user who uses this lib. We will have further discussions to manage types/import errors in one place during the Frankfurt sprint I believe :)
Thank you for the review!